File: 172.Factorial-Trailing-Zeroes.py
# https://leetcode.com/problems/factorial-trailing-zeroes/
#
# algorithms
# Easy (37.09%)
# Total Accepted:    140,770
# Total Submissions: 379,570


class Solution(object):
    def trailingZeroes(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = 0

        while n > 0:
            n /= 5
            res += n

        return res

File: 172.Factorial-Trailing-Zeroes.java
// https://leetcode.com/problems/majority-element/
//
// algorithms
// Easy (37.34%)
// Total Accepted:    154,331
// Total Submissions: 413,341
// beats 100.0% of java submissions

class Solution {
    public int trailingZeroes(int n) {
        if (n < 5) {
            return 0;
        }
        
        return n / 5 + trailingZeroes(n / 5);
    }
}